packages/optimizer/spec/transformers/valid/AmpBoilerplateErrorHandler/adds_error_handler_when_boilerplate_present_esm_enabled/input.html

```diff
@@ -1,3 +1,8 @@
+<!--
+{
+  "esmModuleEnabled": true
+}
+-->
 <!doctype html>
 <html ⚡>
 <head>
```

Comment:
Can we actually use the `.mjs` runtime in this input file? I know that this is not really needed here, but it will probably be needed in the PHP version which also uses these tests.

(could not flag the right line, this file uses the `.js` version a few lines below)

Reply:
out of curiosity, any reason we wouldn't do

```html
<script type="module" src="foo.mjs"></script>
<script nomodule src="foo.js"></script>
```

is that not supported by the php infra?

Reply:
Yes, it is, and with my current version of the code, it is also what is expected to determine the right error handler to use. That's why I'm suggesting using that instead of the regular `v0.js` version of the runtime here.

Reply:
It should be `v0.mjs` though, not `foo.mjs`, as the runtime is the only thing that can be reliably checked.

Reply:
Here's what I have in PHP so far: https://github.com/ampproject/amp-toolbox-php/pull/218/files
As you can see I'm detecting the `v0.mjs` script using an XPath query to decide which version of the errorhandler to use.

The current test in here (which I'd also run off of in PHP) claims that ESM is in use, but does not actually use it. That will break the detection I'm doing of course.

I do think this detection is currently the most correct way, as you could potentially have a user hand-code the ESM version with the configuration `esmModuleEnabled` still being `false`. In that case, the wrong errorhandler will be used in the JS version.

Reply:
👍 done

Reply:
Thanks!